server/controllers/ssotRetrievalController.js

```diff
@@ -77,3 +78,79 @@ exports.renameBot = async (req, res) => {
         console.error(err);
     }
 };
+
+// GET /shareBotWithUser?userid=78d09f66d2ed466cf20b06f7&botId=78d09f66d2ed466cf20b06f7
+exports.shareBotWithUser = async (req, res) => {
+    try {
+        res.set('Content-Type', 'application/json');
+        const { userid } = req.query;
+        const { botId } = req.query;
+
+        const uao = await mongoose.model('userAccessObject').create({
+            AccessLevel: 'ReadWrite',
+            robotId: botId,
+            userId: userid
+        });
+
+
+        res.send(uao);
+    } catch (err) {
+        console.error(err);
+    }
+};
+
+// GET /retrieveMetadataForBot/78d09f66d2ed466cf20b06f7
+exports.retrieveBotMetadata = async (req, res) => {
+    try {
+        res.set('Content-Type', 'application/json');
+        const { botId } = req.params;
+
+        const ssotData = await mongoose.model('SSoT').findById(
+            botId,
+            {
+                robotMetadata: 1
+            }
+        ).exec();
+    } catch (err) {
+        console.error(err);
+    }
+};
+
+
+// GET /createNewBot?userid=78d09f66d2ed466cf20b06f7&botName=NewRobot
+exports.createNewBot = async (req, res) => {
+    try {
+        res.set('Content-Type', 'application/json');
+        const { userid } = req.query;
+        const usableUserId = mongoose.Types.ObjectId(userid);
+        const { botName } = req.query;
+        const nameWithEmptyspace = botName.replace(/\+/g, ' ');
+
+        const ssot = await mongoose.model('SSoT').create({
+            robotMetadata: {
+                starterId: '',
+                robotName: nameWithEmptyspace
+            },
+            elements: []
+        });
+
+        const updatedSsot = await ssot.updateOne({
+            'robotMetadata.robotId': ssot.id
+        }).exec();
+
+        const uao = await mongoose.model('userAccessObject').create({
+            AccessLevel: 'ReadWrite',
+            robotId: ssot.id,
+            userId: usableUserId
+        });
+
+        const returnObj = {
+            robotName: nameWithEmptyspace,
+            robotId: ssot.id
+        };
+
+        res.send(returnObj);
```

Comment:
You used bot instead of robot. Does this have a specific reason? Because so far I think we were always referring to them as robot. I don't mind using bot but I think it should be consistent

Reply:
Fair Point. Hadn't noticed that at all, those code posts came entirely from Erik. At this point also the request to you @kej-jay to use robot instead of bot in the future.
Anyway: Fixed